### PR TITLE
Fix client-server event schemas: move `age`, dedupe fields

### PR DESCRIPTION
**Pull request by turt2live**
Rendered: see 'docs' status check

*Edit: Halfway through this PR the `age` field changed from being "removed" to just "moved" under `unsigned`*.

----

This commit adds support for event schema examples to have references to help reduce the chance of fields being forgotten. This also helps reduce duplication of fields, allowing for a more consistent spec that uses the same values everywhere.

This also removes both `unsigned` and `age` from the examples as per:
* https://github.com/matrix-org/matrix-doc/issues/1524
* https://github.com/matrix-org/matrix-doc/issues/630

Finally, this replaces "localhost" in the examples with an example domain. This is really just a nitpick thing on my part where seeing a "real world" domain is preferred. 

Fixes https://github.com/matrix-org/matrix-doc/issues/1524
Fixes https://github.com/matrix-org/matrix-doc/issues/630
Step towards https://github.com/matrix-org/matrix-doc/issues/1530

### CODE REVIEW

**Comment by ara4n** [comment]
hm, is this just removing age from the examples? or from the schema itself? as `age` as a param turned out to be very much required in the end (for WebRTC timeouts)

**Comment by turt2live** [comment]
From the examples. I'll double check the schemas and make sure the age appears for the call event stuff.

**Comment by ara4n** [comment]
i am still a bit confused about this - in practice, a compliant server is going to show an `unsigned: { age: 1234 }` block in all its /sync responses for a given event. Don't we want to show this in the examples for /sync responses etc (and not just for VoIP events)?

**Comment by turt2live** [comment]
I've gone ahead and just added the `unsigned` field to everything, as per the rationale in https://github.com/matrix-org/matrix-doc/pull/1558/commits/62b1b8b66097f3d8be2f9cc465335f28641d15fb

This is primarily because application services (particularly bridges) have a very good reason to look at this field, alongside clients which do VoIP. It's worth noting that it is not a required field, however it is something that should be provided.

**Comment by turt2live** [comment]
Fixes https://github.com/matrix-org/matrix-doc/issues/1294
(now)

File: event-schemas/examples/m.room.member#invite_room_state [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
-  "age": 242352,
+  "$ref": "m.room.member",
   "content": {
     "membership": "invite",
-    "avatar_url": "mxc://localhost/SEsfnsuifSDFSSEF#auto",
+    "avatar_url": "mxc://domain.com/SEsfnsuifSDFSSEF#auto",
     "displayname": "Alice Margatroid"
   },
   "unsigned": {
```
Comment (uhoreg) [inline event-schemas/examples/m.room.member#invite_room_state]: the `unsigned.age` field gets lost here, which I'm guessing was not intentional

**Pull request merged**